### ISSUE TITLE
Incompatible version of Kotlin on compilation

### ISSUE THREAD

**Issue by as-85** [bluefireteam/audioplayers#1393]
## Checklist
- [x] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [x] I made sure that the issue I am raising doesn't already exist

## Current bug behaviour

When compiling, many of these errors are displayed.

```
e: /Users/./StudioProjects/./build/audioplayers_android/.transforms/255311182dafdd6ac4eb63a5bb962d32/transformed/out/jars/classes.jar!/META-INF/audioplayers_android_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./StudioProjects/./build/volume_controller/.transforms/fd9215dff063f50e68dfae64139c2d8c/transformed/out/jars/classes.jar!/META-INF/volume_controller_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./StudioProjects/./build/package_info_plus/.transforms/79453a5a4234b228dfa8e31803b7fca3/transformed/out/jars/classes.jar!/META-INF/package_info_plus_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./StudioProjects/./build/fluttertoast/.transforms/2ca7d08035b3ebb8fc9a6b18ef5341d7/transformed/out/jars/classes.jar!/META-INF/fluttertoast_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.7.10/d70d7d2c56371f7aa18f32e984e3e2e998fe9081/kotlin-stdlib-jdk8-1.7.10.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation/1.5.0/857678d6b4ca7b28571ef7935c668bdb57e15027/annotation-1.5.0.jar!/META-INF/annotation.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.7.21/a0ba09615c2213d580315e234b3febfea25b757e/kotlin-stdlib-jdk7-1.7.21.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.7.21/1a2eaf898a0dda83037034b10a42053cf8a7caf8/kotlin-stdlib-1.7.21.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: /Users/./.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.7.21/cb02257de8e13e8498f8e2f69f742f2d438e794d/kotlin-stdlib-common-1.7.21.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
```

## Steps to reproduce

<details>
<summary>Code sample</summary>
</details>

## Logs

<details>
<summary>Full Logs</summary>
Flutter doctor:

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.1 22C65 darwin-x64, locale de-DE)
    • Flutter version 3.3.10 on channel stable at /Library/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (5 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/andreas/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.75.0-insider)
    • VS Code at /Applications/Visual Studio Code - Insiders.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • TB132FU (mobile) • HA1RNFWK • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos    • darwin-x64     • macOS 13.1 22C65 darwin-x64
    • Chrome (web)     • chrome   • web-javascript • Google Chrome 109.0.5414.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

## Audio Files/URLs/Sources
-

## Screenshots
-

## Environment information

* audioplayers version: 2.0.0

Platform 1: android
* OS name and version: 12
* Device: Lenovo Tab P11 Pro Gen 2
* build mode: `release`
* error peculiarities (optional): 
* Platforms tested without any issue (optional):

## More information
-

**Comment by Gustl22** [comment]
What `kotlin_version` do you have in your `build.gradle`? Doesn't seem to be related to `audioplayers` as other packages are also marked as incompatible. It seems that your app uses an outdated one.

**Comment by as-85** [comment]
I use the version 1.7.10

**Comment by Gustl22** [comment]
Does the example app work for you?
https://github.com/bluefireteam/audioplayers/tree/main/packages/audioplayers/example

**Comment by as-85** [comment]
I did not write that it does not work. I only pointed out the errors that are displayed during the build process. The function is fine.

**Comment by Gustl22** [comment]
If the example app does not give this warning, it most likely is a fault in your app. Otherwise you can share your project code if you want.

**Comment by as-85** [comment]
Trying it out.

**Comment by Gustl22** [comment]
@as-85 did you test? Thx

**Comment by as-85** [comment]
With the sample project, I get the following error message during the build:

```
Warning: The plugin file_picker requires Android SDK version 33.
For more information about build configuration, see https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
One or more plugins require a higher Android SDK version.
Fix this issue by adding the following to /Users/andreas/Downloads/test/packages/audioplayers/example/android/app/build.gradle:
android {
  compileSdkVersion 33
  ...
}
```

But the build still works.

**Comment by Gustl22** [comment]
So the kotlin thing is definitely something you need to change in your app in the `build.gradle` file. Like done in our [example app](https://github.com/bluefireteam/audioplayers/blob/aee480b99cf4832dbcf2935fe0356119989d5856/packages/audioplayers/example/android/build.gradle#L2).

Regarding the compileSdk, we needed to wait for Flutter to officially support API 33, which is now the case. This will be fixed in the an upcoming release, see #1364
